src/backends/gaston.jl

```diff
@@ -0,0 +1,120 @@
+# https://github.com/mbaz/Gaston.
+const G = Gaston
+const GastonSubplot = G.Plot
+# --------------------------------------------
+# These functions are called by Plots
+# --------------------------------------------
+# Create the window/figure for this backend.
+function _create_backend_figure(plt::Plot{GastonBackend})
+    state_handle = G.nexthandle() # for now all the figures will be kept
+    plt.o = G.newfigure(state_handle)
+end
+
+
+function _before_layout_calcs(plt::Plot{GastonBackend})
+    # Initialize all the subplots first
+    plt.o.subplots = G.SubPlot[]
+
+    grid = size(plt.layout)
+    plt.o.layout = grid
+
+    for sp in plt.subplots
+        gaston_init_subplot(plt, sp)
+    end
+
+    # Then add the series (curves in gaston)
+    for series in plt.series_list
+        gaston_add_series(plt, series)
+    end
+end
+
+function _update_min_padding!(sp::Subplot{GastonBackend})
+    sp.minpad = (20mm, 5mm, 2mm, 10mm)
```

Comment:
I find it highly confusing if a backend sets these minpads

Reply:
This callback is meant for the backend to adjust the padding. If you cannot adjust all the padding right away, because Plots calculates the padding when grid is complex.

With a complex layout, backend cannot align all the subplots itself, Plots has the aligning logic, so it this callback your are supposed to add the padding Plots tells you to. 

For the moment I do not know if it is possible to communicate with gnuplot to estimate the paddings

Reply:
```
l = @layout([a{0.1h}; b [c; d e]])
plot(randn(100, 5), layout = l, t = [:line :histogram :scatter :steppre :bar], leg = false, ticks = nothing)
```
![image](https://user-images.githubusercontent.com/24591123/101157664-cfc40f80-366d-11eb-968e-064061e28641.png)

Look how pyplot aligns http://docs.juliaplots.org/latest/generated/pyplot/#pyplot-ref16 this so well, the axes spines are well aligned even if you put the uneven sized labels (guides).

 It is only possible with this callback, Plots basically calculates needed padding for each subplot in layouts.jl, it is a fairly complex recursive algorithm. First you put all the subplots, tell how much *at least* should each subplot have padding. Then plot tells you how much exactly each subplot would need for alignment.
So basically you should supply the minimal padding, then Plots in this callback tells you how much it should be (more than minimal to align with other subplots)

Reply:
But isn't it also causing unwanted whitespace, when you e.g. set `margin = 0Plots.mm` ?

Reply:
The key word margins are extra margins from users, ideally there should be no need for it at all. But with colorbars and such setting extra margin is a hack, it should've been accounted for by plots